Usability test section (#137)
SW709E19/Report: report/ChapterTest/UsabilityTest.tex

\section{Usability Test}

> This section describes the usability test of the user interface for the systems.

Reviewer: ```suggestion
This section describes the usability test of the system's user interface.
```

The usability test was performed 10-12-2019 with a single test participant.

The participant, Henrik Ruhe, is a full time machine operator at Staby Fliser \& Hegn A/S \cite{StabyFliser}, a small factory in Ulfborg, Denmark. 

> The machine that Henrik operates is a large scale paver block machine which requires both preventive and reactive maintenance.

Reviewer: ```suggestion
The machine that Henrik operates is a large scale paver block machine for which Henrik is responsible for performing preventive- and reactive maintenance of.
```

> The participant has limited technical experience and had no prior knowledge of the system before the test.

Reviewer: ```suggestion
The participant has limited technical knowledge and had no prior experience with the system before the test.
```

The participant was tested in the role of maintenance responsible.

> Due to the participants minimal understanding of English was the test conducted in Danish, however the user interface was not translated.

Reviewer: ```suggestion
Due to the participant's minimal understanding of English, the test was conducted in Danish, however the user interface was not translated.
```

> The test started with informing the participant of the general use case of the system in order for the participant to loosely understand the English terminology, however without showing the user interface beforehand.

Reviewer: Vel ikke kun derfor. Selv hvis han havde styr på Engelsk havde vi nok givet ham forklaringen, right?

> The version of the user interface that was being tested can be seen in \autoref{fig:ui-test}

Reviewer: ```suggestion
The version of the user interface that was being tested can be seen in \autoref{fig:ui-test}.
```

\begin{figure}[H]
	\centering
	\includegraphics[width=1\textwidth]{ChapterTest/images/ui-test-screenshot.png}
	\caption{The user interface used for the testing session.}
	\label{fig:ui-test}
\end{figure}

> The task that the participant was asked to do is as follows:

Reviewer: ```suggestion
The tasks which the participant was asked to do are as follows:
```

\begin{enumerate}
\item Which device is in need of a battery change?
\item How many days and hours before the battery of that same device is depleted?
\item At which sample rate is the device named test-device-2 operating at?
\item When was the last measurement received from that device?
\item Please show the data for the device named test-device.

> \item Where are the device named test-device-3 located?

Reviewer: ```suggestion
\item Where is the device named test-device-3 located?
```

\end{enumerate}

> After the task were complete the participant were asked to answer the following questions:

Reviewer: ```suggestion
After the tasks were completed the participant was asked to answer the following questions:
```

\begin{enumerate}
\item How did you perceive the difficulty of the tasks?
\item Was there anything that was difficult to find?
\item Do you have any comments on the design?
\end{enumerate}

> During the first task were the participant quick to move the cursor above the correct location on the UI, however hesitated to state the name of the device. 

Reviewer: ```suggestion
During the first task, the participant was quick to move the cursor above the correct location on the UI, however hesitated to state the name of the device. 
```

> This could indicate that there were some doubt about the answer, but will be disregarded since it was the first task and the participant had no prior experience with usability testing.

Reviewer: ```suggestion
This could indicate that there was some doubt about the answer, but will be disregarded since it was the first task and it was likely because the participant had no prior experience with usability testing.
```

> For the second task did the participant experience trouble reading the remaining battery time. 

Reviewer: ```suggestion
For the second task, the participant experienced trouble reading the remaining battery lifetime. 
```

> This could be because of the dark red text colour on the dark grey background, or due to the participants limited English understanding.

Reviewer: ```suggestion
This could be because of the dark red text colour on the dark grey background, or due to the participant's limited English understanding.
```

> The third task of finding the sample rate under the configuration window, were the participant not aware of the meaning of the term sample rate, and therefore was given a hint that this value could be found under the configuration of a device. 

Reviewer: Samme problem her som 3. punkt i min generelle feedback. Udsagnsordet "were" kommer ikke først i en "ny sætning" som det gør på dansk.
```suggestion
During the third task of finding the sample rate under the configuration window, the participant was not aware of the meaning of the term sample rate, and was therefore given a hint that this value could be found under the configuration of a device. 
```

> The participant were quick to find the button for configuration and after pressing it able to find the sample rate almost immediately.

Reviewer: ```suggestion
The participant was quick to find the button for configuration and was after pressing it able to find the sample rate almost immediately.
```

The UI was in the beginning intended for multiple actors and not just the maintenance responsible. 
The configuration of a device were intended for the system architect, so that it would not be necessary to access Google Cloud in order to change a devices configuration.

> However, since the configuration window is read only it would be ideal to remove the "Configuration" button for the maintenance responsible part of the system.

Author: Im not sure where this should be placed...

> The fourth task caused issues due to the language barrier and the participant were not able to find the time of the last received measurement. 

Reviewer: ```suggestion
The fourth task caused issues due to the language barrier and the participant was not able to find the time of the last received measurement. 
```

The participant went through all the columns and the configuration and data window before asking for the question to be repeated.

> After the question was repeated and the term "last measurement" was emphasized, the participant were able to find the right row and column where the information were present.

Reviewer: ```suggestion
After the question was repeated and the term "last measurement" was emphasised, the participant was able to find the right row and column where the information was present.
```

The problem indicates that the UI should be translated to the native language of the maintenance responsible at a given factory whenever the system should be deployed. 
The column "Last measurement" could also be named something more appropriate that better explains the content.

> The fifth task were completed with no issues occurring. The participant found the right device, named "test-device", and clicked the "Data" button within reasonable time.

Reviewer: ```suggestion
The fifth task was completed without issues.
The participant found the right device, named "test-device", and clicked the "Data" button within reasonable time.
```

> The sixth and last task were also completed without any notable difficulties.

Reviewer: ```suggestion
The sixth and last task was also completed without any notable difficulties.
```

> For the follow-up questions did the participant show general satisfaction with the usability of the UI and proclaimed that he found the tasks very easy to complete.

Reviewer: ```suggestion
For the follow-up questions, the participant showed general satisfaction with the usability of the UI and proclaimed that he found the tasks very easy to complete.
```

\section{Usability Test}
This section describes the usability test of the system's user interface.
The usability test was carried out on 10-12-2019 with a single test participant. 
Each actor type should be represented in order to test the usability of the UI. 

> However, the maintenance responsible is the only actor who will be using the developed UI.

Reviewer: ```suggestion
However, the maintenance responsible is the only actor who will be using the developed UI and the usability test was therefore limited to this actor alone.
```

The fourth task caused issues due to the language barrier and the participant was not able to find the time of the last received measurement. 
The participant went through all the columns and the configuration and data window before asking for the question to be repeated.
After the question was repeated and the term "last measurement" was emphasised, the participant was able to find the right row and column where the information was present.
The problem indicates that the UI should be translated to the native language of the maintenance responsible at a given factory whenever the system should be deployed. 
The column "Last measurement" could also be named something more appropriate that better explains the content.

The fifth task was completed without issues.
The participant found the right device, named "test-device", and clicked the "Data" button within reasonable time.

The sixth and last task was also completed without any notable difficulties.

For the follow-up questions, the participant showed general satisfaction with the usability of the UI and proclaimed that he found the tasks very easy to complete.

\subsection{Evaluation}

> The usability test gave rise to changes that could be made to offer a better usability.

Reviewer: ```suggestion
The test gave rise to changes that could be made to improve the usability of the system's UI.
```

\begin{itemize}
	\item The text displaying remaining battery lifetime should be made more readable e.g. by changing the colour. 
	\item The dashboard should be translated to the native language of the maintenance responsible of the factory, in this case Danish.
	\item The table columns such as "Last measurement" should be named more appropriately.
\end{itemize}

The third test of finding the sample rate of a device was not relevant for the maintenance responsible and should not have been included in the usability test.

> The UI was during the development of the system extended to include tasks for multiple actors and not just the maintenance responsible. 

Reviewer: ```suggestion
The UI was, during the development of the system, extended to include tasks for multiple actors and not just the maintenance responsible. 
```

> The configuration of a device were intended for the system architect, so that it would not be necessary to access Google Cloud in order to change a device's configuration.

Reviewer: ```suggestion
The configuration of sensor devices was intended for the system architect, so that it would not be necessary to access Google Cloud in order to change a device's configuration.
```

> However, since the configuration window ended up being read only it would be ideal to remove the "Configuration" button from the maintenance responsible's part of the system. 

Reviewer: ```suggestion
However, since the configuration window ended up being read-only it would be ideal to remove the "Configuration" button from the maintenance responsible's part of the system. 
```

> Therefore, if the system was to be extended to include tasks of multiple actors, the UI should be different for each of the actors. 

Reviewer: ```suggestion
In general, if the system is to be extended to include tasks of multiple actors, the UI should be different for each of the actors. 
```

> This could be done by creating user accounts and assigning them different roles such that when logged in, the UI would only display relevant information for each of the actors.

Reviewer: ```suggestion
This could be done by creating user accounts and assigning different roles to them such that when logged in, the UI would only display relevant information for the user.
```
